week-6/Homework/mandatory/1-alarmclock/alarmclock.js

```diff
@@ -1,27 +1,82 @@
-function setAlarm() {}
+// ## How the clock should work
 
-// DO NOT EDIT BELOW HERE
+// When you click the `Set Alarm` button the counter at the top of the screen should change to the number you entered in the `input` field. For example, if the `input` field says `10` then the title should say `Time Remaining: 00:10`.
 
-function setup() {
-  var audio = new Audio("alarmsound.mp3");
+// Every one second the title should count down by one.
+
+// When the `Time Remaining` reaches `00:00` the alarm should play a sound. You can make the sound happen by using `playAlarm()`.
+
+// You can stop the alarm sound by pressing the `Stop Alarm` button.
+
+let reset = document.querySelector('#reset')
 
-  console.log(document.getElementById("set"));
+function setAlarm(){
+  let inputField = document.querySelector('#alarmSet')
+  let title = document.querySelector('#timeRemaining')
+  let time = document.querySelector("#time");
+  let audioEl = document.querySelector("#myAudio")
+  let body = document.querySelector('body')
+  let set = inputField.value*1000
+  let countdown = setInterval(timer, 1000)
+  let setRepeat = document.getElementById("set").addEventListener("click", () => {
```

Comment:
This is a slightly surprising thing to do when you press `set` - can you talk through what this is doing and why?